proposals/text/7462-rag-evaluation.md

```diff
@@ -0,0 +1,243 @@
+- Github Issue or Discussion: https://github.com/deepset-ai/haystack/issues/7398
+
+# Summary
+
+Add a new component to Haystack allowing users interact with the results of evaluating the performance of a RAG model.
+
+
+# Motivation
+
+RAG models are one of them most popular use cases for Haystack. We are adding support for evaluations metrics, but there is no way to present the results of the evaluation.
+
+
+# Detailed design
+
+The output results of an evaluation pipeline composed of `evaluator` components are passed to a `RAGPipelineEvaluation`
+(this is a placeholder name) which stores them internally and acts as an interface to access and present the results.
+
+Example of the data structure that the `RAGPipelineEvaluation` class will receive for initialization:
+
+```python
+
+data = {
+    "queries": {
+        "query_id": ["53c3b3e6", "225f87f7"],
+        "question": ["What is the capital of France?", "What is the capital of Spain?"],
+        "contexts": ["wiki_France", "wiki_Spain"],
+        "answer": ["Paris", "Madrid"]
+    },
+    "metrics":
+        [
+            {"name": "reciprocal_rank", "scores": [0.378064, 0.534964, 0.216058, 0.778642]},
+            {"name": "single_hit", "scores": [1, 1, 0, 1]},
+            {"name": "multi_hit", "scores": [0.706125, 0.454976, 0.445512, 0.250522]},
+            {"name": "context_relevance", "scores": [0.805466, 0.410251, 0.750070, 0.361332]},
+            {"name": "faithfulness", "scores": [0.135581, 0.695974, 0.749861, 0.041999]},
+            {"name": "semantic_answer_similarity", "scores": [0.971241, 0.159320, 0.019722, 1]}
+         ],
+    "pipeline_answers": ["Paris", "Madrid"]
+    },
+
+```
+
+- At least the `query_id` or the `question and context` should be present in the data structure.
+- At least one of the metrics should be present in the data structure.
+- The `pipeline_answers` field is optional, it is used to compare the answers generated by the pipeline with the expected answers.
+
+
+The `RAGPipelineEvaluation` class provides the following methods to different types of users:
+
+Basic users:
+- `evaluation_report()`
+- `comparative_evaluation_summary()`
+
+Intermediate users:
+- `detailed_evaluation_report()`
+- `comparative_detailed_evaluation_report()`
+
+Advanced users:
+- `find_thresholds()`
+- `find_scores_below_threshold()`
+
+
+### Methods description
+An evaluation report that provides a summary of the performance of the model across all queries, showing the
+aggregated scores for all available metrics.
+
+```python
+def evaluation_report():
+```
+
+Example output
+
+```bash
+{'Reciprocal Rank': 0.448,
+ 'Single Hit': 0.5,
+ 'Multi Hit': 0.540,
+ 'Context Relevance': 0.537,
+ 'Faithfulness': 0.452,
+ 'Semantic Answer Similarity': 0.478
+ }
+ ```
+
+A detailed evaluation report that provides the scores of all available metrics for all queries or a subset of queries.
+
+```python
+def get_detailed_scores(query_ids: Union[List[str], str] = "all"):
```

Comment:
Let's use the same names as the ones listed above. Also, let's remove the query ID concept for now and just call them `queries` - how we lookup and cross-ref them is an implementation detail.